### DISCUSSION TITLE
How to create a contract instance from a PublicActions extended WalletClient

### DISCUSSION THREAD

**Discussion by SodiumBoloxide** [wevm/viem#861] [Question]
Given a WalletClient which is extended with PublicActions via the `.extend` method. How would one create a contract instance? If its impossible to do so, then it would be a great feature to add.

Like...
```
const client = createWalletClient({
  account,
  chain,
  transport: http()
}).extend(publicActions);

const contract = getContract({ address, abi, extendedClient });
```

**Comment by tmm** [comment]
Not currently possible to use the extended public actions via the wallet client like this. Worth thinking about more how this would work, etc.

**Reply by SodiumBoloxide** [reply]
Ok thanks
